src/app/state/rewardSlice.ts

```diff
@@ -0,0 +1,429 @@
+        console.error(action.error);
+      })
+
+      // fetchOrderRewards
+      .addCase(fetchOrderRewards.pending, () => {})
+      .addCase(
+        fetchOrderRewards.fulfilled,
+        (state, action: PayloadAction<OrderRewards[]>) => {
+          state.rewardData.ordersRewards = action.payload;
+        }
+      )
+      .addCase(fetchOrderRewards.rejected, (state, action) => {
+        DexterToast.error("Error fetching order rewards ");
+        console.error(action.error);
+      })
+
+      // claimRewards
+      .addCase(claimRewards.pending, (state) => {
+        state.showSuccessUi = false;
+      })
+      .addCase(claimRewards.fulfilled, (state, action) => {
+        state.showSuccessUi = action.payload;
+      })
+      .addCase(claimRewards.rejected, (state) => {
+        state.showSuccessUi = false;
+      });
+    // You can add more cases here
+  },
+});
+
+// Fetches user's NFTs and uses all order receipt NFT addresses from the Adex API
+// to filter out relevant NFTs, returning an array of receiptIdentifiers.
+export const fetchReciepts = createAsyncThunk<
+  string[], // array of receipt identifiers, e.g. "resource_tdx...ayahv#72#"
+  adex.PairInfo[], // argument type
+  {
+    state: RootState;
+  }
+>("rewards/fetchReciepts", async (pairsList, thunkAPI) => {
+  const rdt = getRdtOrThrow();
+  // const walletData = rdt.walletApi.getWalletData();
+  const state = thunkAPI.getState();
+  const accounts = state.radix.walletData.accounts;
+  // Todo support multiple wallets ids
+  const accountAddress = accounts[0].address;
+  // get all NFTs from your wallet
+  const { items } =
+    await rdt.gatewayApi.state.innerClient.entityNonFungiblesPage({
+      stateEntityNonFungiblesPageRequest: {
+        address: accountAddress,
+        // eslint-disable-next-line camelcase
+        aggregation_level: "Vault",
+        // eslint-disable-next-line camelcase
+        opt_ins: { non_fungible_include_nfids: true },
+      },
+    });
+  // console.log("Receipt items: ", items);
+  // load set of orderreceiptAddresses from adex
+  let orderReceiptAddresses: Set<string> = new Set();
+  pairsList.forEach((pairInfo) =>
+    orderReceiptAddresses.add(pairInfo.orderReceiptAddress)
+  );
+  // iterate through all receipts and extract receiptIdentifier
+  const receipts: string[] = [];
+  (items as NonFungibleResource[]).forEach((item) => {
+    if (
+      orderReceiptAddresses.has(item.resource_address) &&
+      item.vaults &&
+      item.vaults.items &&
+      item.vaults.items.length > 0 &&
+      item.vaults.items[0].items &&
+      item.vaults.items[0].items.length > 0
+    ) {
+      for (let i = 0; i < item.vaults.items[0].items.length; i++) {
+        const orderReceiptId = item.vaults.items[0].items[i];
+        receipts.push(`${item.resource_address}${orderReceiptId}`);
+      }
+    }
+  });
+  console.log("Fetched all receipts for account: " + accountAddress, receipts);
```

Comment:
remove
```suggestion
```

Reply:
All the console logs can be removed in a separate PR once the branch has been merged with main and everything is working.